extends Camera2D

@onready var birb_factory = $"../BirbFactory"
@export var scroll_locked:bool = false # –ë–ª–æ–∫–∏—Ä–æ–≤–∫–∞ –ø—Ä–æ–∫—Ä—É—Ç–∫–∏

@onready var viewport_width = self.get_viewport_rect().size.x
var last_mouse = Vector2(0, 0)

func _ready() -> void:
	pass

# –ö–∞–º–µ—Ä–∞ —Å–æ–ø—Ä–æ–≤–æ–∂–¥–∞–µ—Ç –∞–∫—Ç–∏–≤–Ω—É—é –ø—Ç–∏—á–∫—É, –µ—Å–ª–∏ –æ–Ω–∞ –µ—Å—Ç—å
# –ò–Ω–∞—á–µ –ø—Ä–æ–∫—Ä—É—á–∏–≤–∞–µ—Ç—Å—è –ø–æ –¥–µ–ª—å—Ç–∞–º –º—ã—à–∏
func _process(delta: float) -> void:
	var now_mouse = get_global_mouse_position()
	var delta_mouse = now_mouse - last_mouse
	
	if birb_factory.active_birb:
		self.position = birb_factory.active_birb.position
	elif Input.is_mouse_button_pressed(MOUSE_BUTTON_LEFT) and not scroll_locked:
		self.position -= delta_mouse
	
	# –ù–µ –¥–∞–≤–∞—Ç—å –∑–∞–µ–∑–∂–∞—Ç—å –≤ –º–µ—Ä—Ç–≤—ã–µ –∑–æ–Ω—ã –ø—Ä–æ–∫—Ä—É—Ç–∫–∏
	if self.position.x < self.limit_left + viewport_width/2:
		self.position.x = self.limit_left + viewport_width/2
		
	if self.position.x > self.limit_right - viewport_width/2:
		self.position.x = self.limit_right - viewport_width/2

	last_mouse = now_mouse
	
	# –¢–∞—â–∏—Ç—å –∑–∞ –∫–∞–º–µ—Ä–æ–π –º–µ–Ω—é –ø–∞—É–∑—ã üòß
	$"../PauseMenu".position = get_screen_center_position() - get_viewport_rect().size/2
	$"../GameOver".position = get_screen_center_position() - get_viewport_rect().size/2
	
